kernel/sched.c

```diff
@@ -1143,28 +1140,14 @@ void z_impl_k_wakeup(k_tid_t thread)
 	if (!arch_is_in_isr()) {
 		z_reschedule_unlocked();
 	}
-
-	if (IS_ENABLED(CONFIG_SMP) &&
-	    !IS_ENABLED(CONFIG_SCHED_IPI_SUPPORTED)) {
-		z_sched_ipi();
-	}
 }
 
 #ifdef CONFIG_SMP
-/* Called out of the scheduler interprocessor interrupt.  All it does
- * is flag the current thread as dead if it needs to abort, so the ISR
- * return into something else and the other thread which called
- * k_thread_abort() can finish its work knowing the thing won't be
- * rescheduled.
- */
 void z_sched_ipi(void)
 {
-	LOCKED(&sched_spinlock) {
-		if (_current->base.thread_state & _THREAD_ABORTING) {
-			_current->base.thread_state |= _THREAD_DEAD;
-			_current_cpu->swap_ok = true;
-		}
-	}
+	/* NOTE: When adding code to this, make sure this is called
```

Comment:
since this function is now a no-op, and scoped private with z_, can we just delete it?

Reply:
`z_sched_ipi()` is the ISR for IPI. We can possibly delete it but I will let @andyross decide.